samples/csharp/MicrosoftGraphSDK/Assignment.cs

```diff
@@ -134,109 +139,163 @@ public class Assignment
+            public static async Task<EducationAssignment> DeactivateAsync(
+                GraphServiceClient client,
+                string classId,
+                string assignmentId)
             {
-                throw new GraphException($"PostResourceAsync call: {ex.Message}", ex, classId, assignmentId);
+                try
+                {
+                    return await client.Education
+                        .Classes[classId]
+                        .Assignments[assignmentId]
+                        .Deactivate.PostAsync(requestConfig =>
+                        {
+                            requestConfig.Headers.Add(
+                                "Prefer", "include-unknown-enum-members");
+                        });
+                }
+                catch (Exception ex)
+                {
+                    throw new GraphException($"DeactivateAsync call: {ex.Message}", ex, classId, assignmentId);
+                }
             }
-        }
 
         /// <summary>
-        /// Sets up the assignment resources folder
+        /// Create Assignment Feedback resources folder
+        /// Reference :: https://learn.microsoft.com/en-us/graph/assignments-states-transition
         /// </summary>
         /// <param name="client">Microsoft Graph service client</param>
         /// <param name="classId">User class id</param>
-        /// <param name="assignmentId">Assignment id</param>
+        /// <param name="assignmentId">Assignment id in the class</param>
         /// <returns>EducationAssignment</returns>
-        public static async Task<EducationAssignment> SetupResourcesFolder(
-            GraphServiceClient client,
-            string classId,
-            string assignmentId)
-        {
-            try
-            {
-                return await client.Education
-                    .Classes[classId]
-                    .Assignments[assignmentId]
-                    .SetUpResourcesFolder
-                    .PostAsync();
-            }
-            catch (Exception ex)
+        public static async Task<EducationAssignment> SetUpAssignmentFeedbackResourcesFolderAsync(
+                GraphServiceClient client,
+                string classId,
+                string assignmentId)
             {
-                throw new GraphException($"SetupResourcesFolder call: {ex.Message}", ex, classId, assignmentId);
+                try
+                {
+                    var requestInformation = client.Education.Classes[classId].Assignments[assignmentId].SetUpFeedbackResourcesFolder.ToPostRequestInformation();
+                    var requestBody = new Entity(); // create an empty request body
+                    requestInformation.SetContentFromParsable(client.RequestAdapter, "application/json", requestBody);
+
+                    var errorMapping = new Dictionary<string, ParsableFactory<IParsable>> {
+                        { "XXX", ODataError.CreateFromDiscriminatorValue }
```

Comment:
what is the key signifying here?

Reply:
Its the error mapping to be used incase an error is returned from the API. using XXX means that the ODataError class will be used for 400 to 500 response codes.

Reply:
this reply should be a comment in the source code

Reply:
Added the comment "XXX key means its the error mapping to be used incase an error is returned from the API that the ODataError class will be used for 400 to 500 response codes."